demisto_sdk/commands/content_graph/objects/pack_metadata.py

```diff
@@ -0,0 +1,477 @@
+                ]
+            )
+            or any(
+                [
+                    content_items.parsing_rule,
+                    content_items.modeling_rule,
+                    content_items.correlation_rule,
+                    content_items.xdrc_template,
+                ]
+            )
+            else set()
+        )
+        tags |= (
+            {PackTags.DATA_SOURCE}
+            if self._is_data_source(content_items)
+            and marketplace == MarketplaceVersions.MarketplaceV2
+            else set()
+        )
+
+        if self.created:
+            days_since_creation = (
+                datetime.utcnow()
+                - datetime.strptime(self.created, "%Y-%m-%dT%H:%M:%SZ")
+            ).days
+            if days_since_creation <= 30:
+                tags |= {PackTags.NEW}
+            else:
+                tags -= {PackTags.NEW}
+
+        return list(tags)
+
+    def _get_tags_by_marketplace(self, marketplace: str):
+        """Returns tags in according to the current marketplace"""
+        tags: set = set()
+
+        if not self.tags:
+            return tags
+
+        for tag in self.tags:
+            if ":" in tag:
+                tag_data = tag.split(":")
+                if marketplace in tag_data[0].split(","):
+                    tags.update({tag_data[1]})
+            else:
+                tags.update({tag})
+
+        return tags
+
+    def _is_data_source(self, content_items: PackContentItems) -> bool:
+        """Returns a boolean result on whether the pack should considered as a "Data Source" pack."""
+        return (
+            len(
+                [
+                    MarketplaceVersions.MarketplaceV2 in integration.marketplaces
+                    and (integration.is_fetch or integration.is_fetch_events)
+                    for integration in content_items.integration
+                ]
+            )
+            == 1
+        )
+
+    def _get_tags_from_landing_page(self, pack_id: str) -> set:
+        """
+        Build the pack's tag list according to the landingPage sections file.
+
+        Args:
+            pack_id (str): The pack ID.
+
+        Returns:
+            set: Pack's tags.
+        """
+        tags = set()
+        landing_page_sections = get_json(LANDING_PAGE_SECTIONS_PATH)
+        sections = landing_page_sections.get("sections") or []
+
+        for section in sections:
+            if pack_id in landing_page_sections.get(section, []):
+                tags.add(section)
+
+        return tags
+
+    def _get_support_details(self) -> dict:
+        """Gets the support details object for the metadata.
+
+        Returns:
+            dict: The support details object.
+        """
+        support_details = {}
+        if self.url:
+            support_details["url"] = self.url
+        if self.email:
+            support_details["email"] = self.email
+        return support_details
+
+
+def get_author(author, marketplace):
```

Comment:
please move the below methods to the PackMetadata class scope (as they shouldn't be used outside of this scope).